### DISCUSSION TITLE
Files inside Linux Onedrive subfolder aren't deleted

### DISCUSSION THREAD

**Discussion by aseemit** [abraunegg/onedrive#1453] [Q&A] [answered]
Hi: 
When attempting to sync only a particular folder or even everything, I noticed that Linux clients do not delete the local file inside subfolders even if they are deleted from Onedrive, is this s feature or a bug. If its a feature, how to ensure changes propagate in a consistent manner. 

TEST CASES: 
On OSX, delete File in Root of Onedrive
=> Changes propagate to all systems

On Linux, delete files In Root of Onedrive
=> Changes propagate to all systems

On OSX delete file/folder inside a subfolder OneDrive/FolderA/FIle
=> Changes do not propagate on Linux Clients

On Linux delete file/folder inside a subfolder Onedrive/FolderA/File
=> File is deleted from only that Linux host and all other non-Linux hosts

Expected behavior: 
File deleted on any host/any folder with default installation setting should propagate across all hosts/cloud.

**Comment by abraunegg** [comment, marked as answer]
@aseemit 
> I noticed that Linux clients do not delete the local file inside subfolders even if they are deleted from Onedrive, is this s feature or a bug. If its a feature, how to ensure changes propagate in a consistent manner.

What client version are you running on Linux? How are you running the client?

The client absolutely *does* delete files / folders locally when files are deleted from OneDrive.

**Reply by aseemit** [reply]
I have a test setup with CentOS 7/8/Ubuntu 20.04/Debian 10. On CentOS 7/8 compiled the latest release. On Debian 10 used the Testing Repo v2.4.10. On Ubuntu used the repo from the docs. 

ALL CLIENTS ARE RUNNING AS A SERVICE. TESTED MANUALLY AS WELL.

The client deletes files only if removed locally on each Linux Host when the contents are inside a subfolder.

HOST A - CentOS 8
HOME/Onedrive/foldera/filea

If i delete this on this host, it will be deleted on Onedrive and on host A.

HOST B - Ubuntu 20.04
HOME/Onedrive/foldera/filea

No changes happen to this file on Host B even if it was deleted from HostA and Onedrive.

**Reply by abraunegg** [reply]
@aseemit 
> I have a test setup with CentOS 7/8/Ubuntu 20.04/Debian 10. On CentOS 7/8 compiled the latest release. On Debian 10 used the Testing Repo v2.4.10. On Ubuntu used the repo from the docs.

You could have actually provided the client version as per `onedrive --version` ... however:

* CentOS / Fedora RPMS are obsolete - building from source is correct, so you should have the right client version
* Ubuntu / Debian repos are obsolete - building from source is your only option

Only the current release - v2.4.11 or greater is supported.

> No changes happen to this file on Host B even if it was deleted from HostA and Onedrive.

Changes *will* happen. The client, when using a service file (systemd / initd) or `--monitor` will monitor for changes made on OneDrive and apply them every 5 minutes are per this note in the readme:

> * While local changes are uploaded right away, remote changes are delayed until next automated sync cycle when using --monitor

This activity is governed by the 'monitor_interval' value in the configuration file: https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#monitor_interval

Perhaps you are expecting *instant* change - this is not how the client operates.

If you *think* you have a bug or issue, raise a new issue ticket filling in all the required details using a client version that is supported.

**Reply by aseemit** [reply]
Its not an instant change issue. It just doesn't detect the change at all. I have let it sit for hours and the client just doesn't notice that the file was deleted. It only notices it when the file exists on the root[OneDrive] Folder. 

And its not a matter of the version either because the same thing happens on CentOS7/8 which are using compiled versions of v2.4.11. So this seems to be an issue with not accurately tracking changes inside subfolders. I can create a ticket if you recommend going that route.

**Reply by aseemit** [reply]
I guess another helpful bit of information is that even turning on verbose logging, the output for that particular file is. 

"The file has not changed."

**Reply by abraunegg** [reply]
@aseemit 
Please follow the correct support process using a supported client version to validate your issue and potential bug.

**Reply by aseemit** [reply]
FIX: 
<img width="1591" alt="image" src="https://user-images.githubusercontent.com/26338922/118333901-8f00da00-b4da-11eb-825e-ddc67436d4a9.png">

I didn't know that sync operations were different for NCD vs regular OD. But removing USL4 from src/main.d was the fix. I think you said you don't have a way to test on NCD deployments so, just wanted to let you know what was the change I made .
